diff --git a/kernel/core_hook.c b/kernel/core_hook.c
index 83413b47..70f7cdf7 100644
--- a/kernel/core_hook.c
+++ b/kernel/core_hook.c
@@ -134,6 +134,10 @@ void escape_to_root(void)
 	memcpy(&cred->cap_ambient, &profile->capabilities.effective,
 	       sizeof(cred->cap_ambient));
 
+	// set ambient caps to all-zero
+	// fixes "operation not permitted" on dbus cap dropping
+	memset(&cred->cap_ambient, 0, sizeof(cred->cap_ambient));
+
 	// disable seccomp
 #if defined(CONFIG_GENERIC_ENTRY) &&                                           \
 	LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
